internal/handlers/pg/pgdb/settings.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pgdb
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+
+	"github.com/jackc/pgx/v4"
+	"golang.org/x/exp/slices"
+
+	"github.com/FerretDB/FerretDB/internal/fjson"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+const (
+	// Internal collections prefix.
+	collectionPrefix = "_ferretdb_"
+
+	// Settings table name.
+	settingsTableName = collectionPrefix + "settings"
+
+	// PostgreSQL max table name length.
+	maxTableNameLength = 63
+)
+
+// CreateSettingsTable creates FerretDB settings table if it doesn't exist.
+func (pgPool *Pool) CreateSettingsTable(ctx context.Context, db string) error {
+	tables, err := pgPool.tables(ctx, db)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	if slices.Contains(tables, settingsTableName) {
+		return ErrAlreadyExist
+	}
+
+	sql := `CREATE TABLE ` + pgx.Identifier{db, settingsTableName}.Sanitize() + ` (settings jsonb)`
+	_, err = pgPool.Exec(ctx, sql)
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	settings := must.NotFail(types.NewDocument("collections", must.NotFail(types.NewDocument())))
+	sql = fmt.Sprintf(`INSERT INTO %s (settings) VALUES ($1)`, pgx.Identifier{db, settingsTableName}.Sanitize())
+	_, err = pgPool.Exec(ctx, sql, must.NotFail(fjson.Marshal(settings)))
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	return nil
+}
+
+// GetTableName returns the name of the table for given collection or error.
+func (pgPool *Pool) GetTableName(ctx context.Context, db, collection string) (string, error) {
```

Comment:
Here also: the function not only creates the enigmatic settings table but also updates it.

Reply:
I've added more details on the function behavior.